Sources/TuistGenerator/Generator/SchemesGenerator.swift

```diff
@@ -119,9 +119,22 @@ final class SchemesGenerator: SchemesGenerating {
         if fileHandler.exists(sharedPath) { try fileHandler.delete(sharedPath) }
     }
     
-    private func createDefaultScheme(target: Target, project: Project, buildConfiguration: String) -> Scheme {
+    private func createDefaultScheme(target: Target, project: Project, buildConfiguration: String, graph: Graphing) -> Scheme {
         let targetReference = TargetReference.project(path: project.path, target: target.name)
-        let testTargets = target.product.testsBundle ? [TestableTarget(target: targetReference)] : []
+        
+        let testTargets: [TestableTarget]
+        
+        if target.product.testsBundle {
+            testTargets = [TestableTarget(target: targetReference)]
+        } else {
+            // Find all test targets that depend on this target
+            testTargets = graph.targets
```

Comment:
OK this is actually wrong, it contains all targets in the project, not just those that depend on this target. Will need to add an extra filter step.